addOns/pscanrulesBeta/src/main/java/org/zaproxy/zap/extension/pscanrulesBeta/CacheableScanRule.java

```diff
@@ -426,9 +426,17 @@ public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
                                     msg.getRequestHeader().getURI());
                             lifetimeFound = true;
                             lifetimesFound++;
-                            // get the portion of the string after "s-maxage="
+                            // get the portion of the string after "s-maxage=" up to a comma or the
+                            // end
+                            int sMaxAgeLen = "s-maxage=".length();
```

Comment:
IMO it would be better to extract a method and handle the parsing properly for all directives (`maxage` is doing something similar).

Reply:
Thanks. I didn't really look around because I hadn't hit it anywhere else, even after a ton of proxying. But I don't mind making it a bit more robust across the board.